src/Core/ImageFunction/ResampleSegmentedImage/Code.cxx

```diff
@@ -17,81 +17,38 @@
  *=========================================================================*/
 #include "itkImage.h"
 #include "itkImageFileReader.h"
-#include "itkIdentityTransform.h"
+#include "itkImageFileWriter.h"
 #include "itkLabelImageGaussianInterpolateImageFunction.h"
 #include "itkNearestNeighborInterpolateImageFunction.h"
 #include "itkResampleImageFilter.h"
 
-#include "itkCustomColormapFunction.h"
-#include "itkScalarToRGBColormapImageFilter.h"
-#include "itkRGBPixel.h"
-#include "itkMersenneTwisterRandomVariateGenerator.h"
-
-#ifdef ENABLE_QUICKVIEW
-#  include "QuickView.h"
-#endif
-
-#include <iostream>
-
-using InputPixelType = unsigned short;
-using ImageType = itk::Image<InputPixelType, 2>;
-using RGBPixelType = itk::RGBPixel<unsigned char>;
-using RGBImageType = itk::Image<RGBPixelType, 2>;
-
-using ColormapType = itk::Function::CustomColormapFunction<ImageType::PixelType, RGBImageType::PixelType>;
-
-static void
-CreateRandomColormap(unsigned int size, ColormapType::Pointer colormap);
-
 int
 main(int argc, char * argv[])
 {
-  if (argc != 3)
+  if (argc != 6)
   {
-    std::cerr << "Usage: " << argv[0] << " inputImageFile pixelSize" << std::endl;
+    std::cerr << "Usage: " << argv[0] << " inputImageFile spacingFraction sigmaFraction outputImageFileLabelImageInterpolator outputImageFileNearestNeighborInterpolator" << std::endl;
 
     return EXIT_FAILURE;
   }
-  double spacing = std::stod(argv[2]);
-
-  using ReaderType = itk::ImageFileReader<ImageType>;
-
-  // Identity transform.
-  // We don't want any transform on our image except rescaling which is not
-  // specified by a transform but by the input/output spacing as we will see
-  // later.
-  // So no transform will be specified.
-  using TransformType = itk::IdentityTransform<double, 2>;
+  const char * inputImageFile = argv[1];
+  double spacingFraction = std::stod(argv[2]);
+  double sigmaFraction = std::stod(argv[3]);
+  const char * outputImageFileLabelImageInterpolator = argv[4];
+  const char * outputImageFileNearestNeighborInterpolator = argv[5];
 
-  using GaussianInterpolatorType = itk::LabelImageGaussianInterpolateImageFunction<ImageType, double>;
-  using NearestNeighborInterpolatorType = itk::NearestNeighborInterpolateImageFunction<ImageType, double>;
+  constexpr unsigned int Dimension = 2;
+  using PixelType = unsigned char;
+  using ImageType = itk::Image<PixelType, Dimension>;
```

Comment:
To improve readability, I think it is better to avoid the global type definitions and prefer local ones. The meaning is known without needing to reference multiple sections of code.